pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -244,6 +279,24 @@ private void updateInstanceConfigIfNeeded(String host, int port) {
       needToUpdateInstanceConfig = true;
     }
 
+    // Update instance config with environment properties
+    if (_pinotEnvironmentProvider != null) {
+      // Retrieve failure domain information and add to the environment properties map
+      String failureDomain = _pinotEnvironmentProvider.getFailureDomain();
+      environmentProperties.put(CommonConstants.INSTANCE_FAILURE_DOMAIN, failureDomain);
+
+      // Fetch existing environment properties map from instance configs
+      Map<String, String> existingEnvironmentConfigsMap = instanceConfig.getRecord().getMapField(
+          CommonConstants.ENVIRONMENT_IDENTIFIER);
+
+      if (existingEnvironmentConfigsMap != null && !existingEnvironmentConfigsMap.equals(environmentProperties)) {
+        instanceConfig.getRecord().setMapField(CommonConstants.ENVIRONMENT_IDENTIFIER, environmentProperties);
+        LOGGER.info("Updated instance config for instance: {} with environment specific properties: {}",
```

Comment:
The log message here (says "Updated...") and the one below (says "Updating....") are misleading in sequence. Suggest this change
```suggestion
        LOGGER.info("Adding env properties for instance {}  with environment specific properties: {}",
```